no-vary-query.md

```diff
@@ -0,0 +1,264 @@
+
+## Use cases
+
+We've found several examples of scenarios where data is carried by the query string, but authors would prefer if it does not affect caching:
+
+### Carrying data that is or can be processed by client-side script only
+
+Some examples:
+
+- Analytics data, e.g. what marketing campaign led the user to the destination.
+- Where the user should be redirected to after some action is complete, e.g. a login process.
+- Some detail to highlight on the page, e.g. map coordinates to zoom to or a product variant to emphasize.
+
+Although in theory such data could be passed via the fragment instead of query string, prevailing practice often uses the query string. Additionally, preloading changes the calculus here: a page might want to do server-side processing if possible, but client-side processing if doing so would unlock the fast-loading benefits of preloading.
+
+Another use case for this is entirely client-side-rendered skeleton pages. However, we're not focused on such cases, since they often use the path component to carry the variable information. See [future work on `No-Vary-Path`](#no-vary-path) for more discussion.
+
+### Avoiding unnecessary cache mismatches due to inconsistent referrers
+
+TODO E.g. other sites don't care; legacy server upgrades; ...
+
+### Carrying data not yet determined at the time of preloading
+
+The most prominent example here is analytics. Consider:
+
+```html
+<script type="speculationrules">
+{
+  "prerender": [{
+    "source": "list",
+    "urls": ["/articles/new-underwater-phone"]
+  }]
+}
+</script>
+
+<a href="/articles/new-underwater-phone?via=heroimage">
+  <img src="underwaterphone.jpg" alt="A phone, underwater!">
+</a>
+
+<a href="/articles/new-underwater-phone?via=headline">New underwater phone, just released!</a>
+```
+
+When the page loads, we don't know yet whether the user will click on the hero image or the headline link. So it's useful to prerender the page with no query parameter, and then later [fill in that information](#interaction-with-prerendering-activation) once the user clicks on a link.
+
+_Note: analytics, where you carry generic data such as the marketing campaign or referring source, is different than [navigational tracking](https://privacycg.github.io/nav-tracking-mitigations/#terminology), which passes along user identifiers. Nothing about this proposal is intended to aid navigational tracking, and we expect it to be fully compatible with future and [currently-deployed mitigations](https://privacycg.github.io/nav-tracking-mitigations/#deployed-mitigations) for navigational tracking._
+
+### Customizing behavior when preloading
+
+TODO
+
```

Comment:
This might be rather niche but we could also consider cases where it impacts server-side processing in a way that doesn't affect the resulting resource, e.g., to request load balancing to a particular instance/pool or enable server-side debug logging for a particular request.

While these could be indicated with a request header instead, it's not always convenient either client-side (to add request headers to some types of request, such as link navigations, form submissions and subresource requests) or server-side (server software might be written in such a way as to make policy decisions before request headers are received).

Reply:
I like this expansion of the framing, and think it'll be pretty compelling! I've tried to incorporate it; PTAL.